src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Services/OrganisationsService.cs

```diff
@@ -0,0 +1,115 @@
+        /// <param name="memoryCache">the memory cache</param>
+        /// <param name="logger">the logger</param>
+        /// <param name="generalSettings">the settings which contains the url to the json organisation file</param>
+        public OrganisationsService(HttpClient httpClient, IMemoryCache memoryCache, ILogger<OrganisationsService> logger, IOptions<GeneralSettings> generalSettings)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+            _organisationListLocation = new Uri(generalSettings.Value.GetOrganisationRepositoryLocation);
+
+            _memoryCache = memoryCache;
+            _cacheEntryOptions = new MemoryCacheEntryOptions()
+                .SetPriority(CacheItemPriority.Normal)
+                .SetAbsoluteExpiration(new TimeSpan(1, 0, 0));
+        }
+
+        /// <inheritdoc/>
+        public async Task<string> LookupOrg(string orgNumber)
+        {
+            Organisation organisation;
+
+            if (!_memoryCache.TryGetValue(_orgDictionaryCacheKey, out Dictionary<string, Organisation> organisationDictionary))
+            {
+                organisationDictionary = await HarvestOrgs();
+            }
+
+            organisationDictionary.TryGetValue(orgNumber, out organisation);
+
+            return organisation?.Org;
+        }
+
+        private async Task<Dictionary<string, Organisation>> HarvestOrgs()
+        {
+            try
+            {
+                HttpResponseMessage response = await _httpClient.GetAsync(_organisationListLocation);
+
+                response.EnsureSuccessStatusCode();
+
+                JObject orgs = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+                orgs = (JObject)orgs.GetValue("orgs");
+
+                Dictionary<string, Organisation> organisationHarvest = JsonConvert.DeserializeObject<Dictionary<string, Organisation>>(orgs.ToString());
+
+                if (organisationHarvest.Count == 0)
+                {
+                    throw new OrganisationHarvestException($"Organisation list is empty!");
+                }
+
+                Dictionary<string, Organisation> organisationDictionary = new Dictionary<string, Organisation>();
+
+                foreach (KeyValuePair<string, Organisation> element in organisationHarvest)
+                {
+                    Organisation candidateOrganisation = element.Value;
+                    candidateOrganisation.Org = element.Key;
+
+                    string orgNumber = candidateOrganisation.OrgNumber;
+                    string org = candidateOrganisation.Org;
+
+                    if (string.IsNullOrEmpty(orgNumber))
+                    {
+                        _logger.LogWarning($"Organisation {candidateOrganisation.ToString()} is missing orgNumber and will not be part of orgNumberToOrganisation map!");
+                        throw new OrganisationHarvestException($"Unable to harvest org number for organisation {candidateOrganisation.Org}");
```

Comment:
An exception here would stop the processing. Let's see if just logging it is enough for now.